woocommerce/woocommerce-ios
# Hack Week: Store Order Notifications

# Note

Apologies for the Big PR, as I did not know if this project was going to be shipped successfully, I was not able to break it efficiently beforehand 🙏 

# Why

For my Hack Week project, I decided to explore providing a richer UI for order notifications. 

# How

- Adds a Notification Content Extension Target. This is where the notification UI runs as it is not part of the main app.

- Read credentials from the keychain: This is needed to fire requests. We only care about wpcom credentials as push notifications only work when jetpack is active and connected.

- When a user long presses on an order notification, the extension:
    - Loads the notification object from the server
    - Loads the order object from the server
    - Formats and renders the richer UI.
    
# How to review

The code lives in three different files:
 - `OrderNotificationViewController`
 - `OrderNotificationViewModel`
 - `OrderNotificationView`
 
The other files are just assets, configurations, and a required storyboard as an entry point.

# How to test

- Download the notification extensions development profile from our development-profile bundle.
- Run the app into the device, and make sure notifications are enabled
- Close the app and create an order via web to the store.
- Wait for the push notification to arrive and long press on it.
- See the custom UI for the order notification.

# Screenshots
Light | Dark 
---- | ----
![light](https://github.com/woocommerce/woocommerce-ios/assets/562080/7cae8db1-ea3a-42f9-b27f-02de1ae20ebf)| ![dark](https://github.com/woocommerce/woocommerce-ios/assets/562080/3fca5ab8-07dd-4544-ba7f-d53f79fdd4cd)


    
---
